optuna/_deprecation.py

```diff
@@ -0,0 +1,92 @@
+import functools
+import inspect
+import textwrap
+from typing import Any
+from typing import Callable
+import warnings
+
+from optuna._experimental import _get_docstring_indent
+from optuna._experimental import _validate_version
+
+
+_DEPRECATION_NOTE_TEMPLATE = """
+
+.. note::
+    Deprecated in v{ver}. This feature will be removed in the future. See
+    https://github.com/optuna/optuna/releases/tag/v{ver}.
+"""
+
+
+def deprecation(version: str, name: str = None) -> Any:
+    """Decorate class or function as deprecation.
+
+    Args:
+        version: The first version that supports the target feature.
```

Comment:
Seems like this text is copied from the experimental decorator but should probably be updated.